src/pull-request-core.ts

```diff
@@ -0,0 +1,234 @@
+                    head: sourceBranch,
+                    base: destinationBranch,
+                    maintainer_can_modify: maintainerCanModify,
+                    draft
+                })
+                .then(({ data }) => data)
+        );
+    }
+
+    /**
+     * Pushes a commit of treeSha to commitBranch and returns the response.
+     */
+    function pushCommit(
+        owner: string,
+        repo: string,
+        commitBranch: string,
+        commitBranchSha: string,
+        treeSha: string,
+        authorName: string,
+        authorEmail: string,
+        commitMessage: string
+    ): Promise<GitUpdateRefResponse> {
+        return Promise.resolve(
+            octokit.git
+                .getCommit({
+                    owner,
+                    repo,
+                    commit_sha: commitBranchSha
+                })
+                .then(({ data }) =>
+                    octokit.git.createCommit({
+                        owner,
+                        repo,
+                        message: commitMessage,
+                        tree: treeSha,
+                        author: {
+                            name: authorName,
+                            email: authorEmail,
+                            date: new Date().toISOString()
+                        },
+                        parents: [data.sha]
+                    })
+                )
+                .then(({ data }) =>
+                    // update the branch ref to point to the created commit
+                    octokit.git.updateRef({
+                        owner,
+                        repo,
+                        ref: 'heads/' + commitBranch,
+                        sha: data.sha
+                    })
+                )
+                .then(({ data }) => data)
+        );
+    }
+
+    /**
+     * Creates a tree from the provided commitBranchSha ref for sourceFiles and returns the sha.
+     */
+    function createTree(
+        owner: string,
+        repo: string,
+        commitBranchSha: string,
+        sourceFiles: SourceFile[]
+    ): Promise<string[]> {
+        const tree = sourceFiles.map<Octokit.GitCreateTreeParamsTree>(
+            ({ path, content }) => ({
+                path,
+                content,
+                type: 'blob',
+                mode: '100644'
+            })
+        );
+        return Promise.resolve(
+            octokit.git
+                .createTree({
+                    owner,
+                    repo,
+                    tree,
+                    base_tree: commitBranchSha
+                })
+                .then(({ data }) => [data.sha, commitBranchSha])
+        );
+    }
+
+    /** Creates a a commitBranch from baseBranch and returns the sha. */
+    function createCommitBranch(
+        owner: string,
+        repo: string,
+        baseBranch: string,
+        commitBranch: string
+    ): Promise<string> {
+        return Promise.resolve(
```

Comment:
`Promise.resolve()` is not needed when you already have a Promise at hand.

Reply:
Derp. Thanks.